feat: Add video tag support for context

NicholasBoll opened the pull request:
This is a very simple implementation of adding video support. It recognizes a video link in the context and makes a `video` tag instead of just a URL. It follows the same implementation of the `img` tag. Not all video extensions are supported - just the ones I tested. More could easily be added.

It looks like this:
![screen shot 2018-01-20 at 9 57 33 am](https://user-images.githubusercontent.com/338257/35185899-7e367f3a-fdc8-11e7-9052-cf5e1f30e268.png)

NicholasBoll commented on src/components/test/context.jsx:
```diff
@@ -7,6 +7,7 @@ import styles from './test.css';
 
 const cx = classNames.bind(styles);
 
+const videoRegEx = /(?:mp4|webm)$/i;
```
More extensions could be added here. These are common and non-ambiguous.

NicholasBoll commented on src/components/test/context.jsx:
```diff
@@ -24,6 +25,19 @@ class TestContext extends Component {
+
+    return (
+      <video controls src={ linkUrl } className={ cx('video') }>
+        <track kind='captions' />
```
The accessibility checker flags video tags without captions even though there is caption source file. I didn't know what would be better - disabling the lint rule or fooling it. Either way there are no captions.

NicholasBoll commented on src/components/test/test.css:
```diff
@@ -264,3 +264,9 @@
 }
+
+.video {
+  display: block;
```
Same as img CSS. Makes the video scale responsively like the image tag does.

adamgruber replied:
This should be combined with the `.image` class definition since they are the same.

NicholasBoll commented:
The video support is simple and automatic. I don't know if the maintainer wants a more formal context type to opt in to more video features (captions, poster, multiple sources, etc). This change works well for adding video as context for failures of web UI tests (Cypress or Selenium)

adamgruber commented:
@NicholasBoll Thanks for the PR! This looks good overall, just a comment about the CSS. Also, could you add tests for this?

NicholasBoll commented:
Sure thing. Thanks for taking a look!

NicholasBoll commented on test/spec/components/test/context.test.jsx:
```diff
@@ -134,6 +136,45 @@ describe('<TestContext />', () => {
+      });
+      expect(wrapper).to.have.className('test');
+      expect(snippet).to.have.lengthOf(0);
+      expect(wrapper).to.have.descendants('video');
```
I did `expect(wrapper).to.have.descendants('video')` instead of `expect(video).to.have.lengthOf(1)` for the failure case.

```js
expect(video).to.have.lengthOf(1);
// failure: Expected 0 to be 1

expect(wrapper).to.have.descendants('video');
// failure: expected <TestContext /> to have descendants 'video' (and then dumps the HTML for easy comparison)
```

adamgruber replied:
I like this change but it looks like we could keep the length check using:
```js
expect(wrapper).to.have.exactly(1).descendants('video');
```

NicholasBoll replied:
I like it

NicholasBoll commented:
@adamgruber I've fixed the CSS, added an anchor tag in the video tag (in case the browser doesn't render the video tag - the anchor will render instead). I also added tests, but I changed the style slightly for better failure cases.

adamgruber commented on test/spec/components/test/context.test.jsx:
```diff
@@ -134,6 +136,45 @@ describe('<TestContext />', () => {
+      expect(wrapper).to.have.descendants('video');
+      expect(wrapper).to.have.descendants('a.test-video-link');
+      expect(videoLink).to.have.attr('href', 'http://test.url.com/testvideo.mp4');
+      videoLink.simulate('click', { stopPropagation: noop });
```
What's the reason for the simulated click here?

NicholasBoll replied:
I don't know, it was part of the img one I copied.

adamgruber replied:
Haha. Ok fair enough. I don't remember why I added that.

NicholasBoll commented:
@adamgruber I've made the suggested changes. Thank you for taking the time to make this PR better
